Voice Moderation Project Viewer

Geenz opened the pull request:
as of = 2025-12-12 for PV deploy
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project_Voice_Moderation%23c5c9b1f-2025-12-11
cohort = Voice Moderation
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project_Voice_Moderation%23c5c9b1f-2025-12-11
desired = 0
relnotes:

### Second Life Voice Moderation Alpha

Have you heard? We're adding Voice moderation along with hosting our own WebRTC voice solution!  Land owners will soon be able to moderate voice users on their own land.  This Alpha viewer will have very limited availability, but we'd love for the community to try it out and give us feedback early.

#### How can you help?

The Second Life Project Voice Moderation viewer introduces the ability to moderate spatial voice chat in regions configured to use webRTC voice.

**Who can moderate:**

- Region owners and estate managers can moderate voice on parcels configured to use the region voice channel ('Restrict Voice to this parcel' disabled)
- Parcel owners can moderate voice on parcels configured to use their own parcel voice channel ('Restrict Voice to this parcel' enabled)
- If a parcel is owned by a group and uses its own voice channel, any group member with the 'Moderate Group Chat' ability can moderate

**How to moderate:**

- Go to the location you wish to moderate, and ensure you are connected to spatial voice
- Open the 'Conversations' floater, and right click on any username in the 'Nearby chat' list.  If you have moderation ability, the 'Moderation options' menu will appear
- Select the appropriate moderation action, which might include:
   - Mute everyone: mute all avatars connected to the spatial voice channel.  Note that this setting is 'sticky': late-joining avatars will also be muted.
   - Unmute everyone: unmute all avatars connected to the spatial voice channel, so that they may speak if they wish.
   - Mute this participant: mute the selected avatar
   - Unmute this participant: unmute the selected avatar

**Where to test moderation:**

Moderation can currently be tested on any webRTC-enabled region, which currently means any region on the 'Second Life Preflight' channel.

For those who don't have a webRTC-enabled region that they have powers over, some public test parcels have been set up at https://maps.secondlife.com/secondlife/WebRTC%20Voice%201/114/225/23

The two parcels are owned by a publicly joinable group, which has been configured to allow for any member to moderate voice chat.  Both parcels are configured to use their own parcel voice channel.  Click the green orb for instructions.
The main parcel in 'WebRTC Voice 1' uses the region channel, and can only be moderated by the region owner or an estate manager.

#### Help Us Keep Improving!
Help us shape the future of Second Life! Found a bug or have feedback? Please submit it via our [[Feedback Portal](https://feedback.secondlife.com/)](https://feedback.secondlife.com/).

Thanks for being a vital part of the Second Life journey. We can’t wait to see what you create with these latest enhancements!

LLGuru committed:
#2997 The 'Reset Skeleton' option is missing in the right-click menu

akleshchev committed:
#2639 Reapplied GPU benchmark fix

First test returns quarter to a half the throughput, do two tests.
May be caused by driver, may be some 'energy saving', but not important
enough to spend time investingating. It was working the same way prior
to ExtraFPS, but viewer was running an extra CPU test that 'preheated'
the system.

Also increasing minimum throughput as numerous new features, like
mirrors and pbr were added and requirements are now higher.

callumlinden committed:
First commit for a second experiment, investigating how nearby voice/chat moderation might work

callumlinden committed:
Remove separator bar for Nearby Voice

callumlinden committed:
Improve robustness of when moderator options appear and add some initial code for muting indivudual / everyone via the capability

callumlinden committed:
Merge branch 'develop' of https://github.com/secondlife/viewer into callum/voice-moderation-2

callumlinden committed:
Merge branch 'develop' of https://github.com/secondlife/viewer into callum/voice-moderation-2

maxim-productengine committed:
#4013 Update voice moderator options; show notifications when muted; add stub code for info request

maxim-productengine committed:
#4013 add simple voice moderation permission check

maxim-productengine committed:
Merge branch 'develop' into maxim/voice-moderation

maxim-productengine committed:
Show moderator options only on webrtc region

maxim-productengine committed:
Ignore muted flags from non-primary voice server

maxim-productengine committed:
#4994 remove redundant moderator_id key

maxim-productengine committed:
#4995 change muted/unmuted alerts to non-modal toast

maxim-productengine committed:
Toggle off 'Speak' button when muted by moderator

maxim-productengine committed:
#5018 add webrtc connection statistics

maxim-productengine committed:
#5018 mac build fix

maxim-productengine committed:
#5055 don't show moderate menu if the user is not parcel owner within a parcel with restricted voice

maxim-productengine committed:
Merge branch 'develop' into maxim/voice-moderation

maxim-productengine committed:
Merge branch 'develop' into maxim/voice-moderation

maxim-productengine committed:
#5088 Hide 'Moderation options' menu when disconnected from spatial voice

maxim-productengine committed:
Merge branch 'release/2026.01' into maxim/voice-moderation

# Conflicts:
# indra/newview/llviewerstats.h

maxim-productengine committed:
#5129 Hide moderator options for participants who are not connected to the same voice space

maxim-productengine committed:
#5122 update version format

maxim-productengine committed:
#5122 update version format #2

jellyfish-linden commented:
| Project Voice Moderation 26.1.0.20139269477 | Result | Comment |
|---|---|---|
| secondlife/iqa#580 Acceptance Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**

- Updating from project viewer -> release viewer does not override project viewer install dir. This is expected.
- Mac had trouble downloading the old target viewer per updater step part 4.3 and 4.4. @Dan-Linden to investigate (not a blocker)
- We should expand the field so it can handle more than the number of character it currently has for version numbers in VVM (bug incoming)


Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | SecureAge - Malicious<br />MaxSecure - Trojan.Malware.300983.susgen |

kylelinden reviewed (approved):
Voice Moderation Alpha approved.
viewer = 2026.1.0.20108649989
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project_Voice_Moderation%23c5c9b1f-2025-12-11
cohort = Voice Moderation
desired = 0

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Project_Voice_Moderation%23c5c9b1f-2025-12-11

RyeMutt committed:
secondlife/viewer#5083 Fix external editor default open handling failing to work on mac and windows

akleshchev committed:
#3612 "Copy SLURL" from Favorites bar not working #3

akleshchev committed:
#5086 Update links from settings to https

akleshchev committed:
#5275 Optimize hasParcelLandmark

Implemented result caching.
hasParcelLandmark can get repeatedly called when performing operations
on landmarks en masse, which was causing repeated inventory searches
and leads to stalls with large collections of landmarks.

akleshchev committed:
#5275 Make landmark search by position cheaper

by stopping after first item was found.

akleshchev committed:
#4267 Slight logging improvement for mute list

akleshchev committed:
#5084 Adjust watchdog to avoid false-positives

And increase allowed time to be more in tune with disconnects.

maxim-productengine committed:
#5304 fix UI inconsistencies related to maturity setting in Legacy Search floater

akleshchev committed:
#5086 Restore guidebook to use http

roxanneskelly committed:
Fix failure to reconnect after disconnect and occasional dropout issue (#5322)

* Fix failure to reconnect after disconnect and occasional dropout issue

We were occasionally seeing dropouts which may have been caused by ICE
renegotiate requests.  The code is there to reconnect in that case,
but there were a few bugs, some of which were likely due to the webrtc upgrade.

Also, we were seeing failures to reconnect after voice server restart.

There were some issues with the PTT button that came up after the above issue was fixed.

* Added a clarification as part of CR

* We need to set mute state for p2p/adhoc/group calls as well

akleshchev committed:
#5046 Fix a long freeze when fetching inventory

akleshchev committed:
#2639 Fix benchmarked render level being too high for Apple Silicon

After benchmark fixes, render level jumped from lowest to high+, cap it at med+ until shadows get fixed to have less impact

marchcat committed:
#2997 Fix Reset Skeleton not working on animesh objects

maxim-productengine committed:
Merge branch 'release/2026.01' into project/voice_moderation

Dan-Linden commented:
| Project Voice Moderation 26.1.0.21681382563 | Result | Comment |
|---|---|---|
| secondlife/iqa#607 Updater Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**
Known issue: Mac Bugsplats are not decoded https://github.com/secondlife/viewer/issues/4206

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | SecureAge - Malicious<br />MaxSecure - Trojan.Malware.300983.susgen |

Geenz commented:
Closing in favor of shipping in 26.2.
